lib/parser.dart

```diff
@@ -1229,13 +1229,30 @@ class _Parser {
         combinatorType = TokenKind.COMBINATOR_PLUS;
         break;
       case TokenKind.GREATER:
+        // Parse > or >>>
         _eat(TokenKind.GREATER);
-        combinatorType = TokenKind.COMBINATOR_GREATER;
+        if (_maybeEat(TokenKind.GREATER)) {
+          _eat(TokenKind.GREATER);
+          combinatorType = TokenKind.COMBINATOR_DEEP;
+        } else {
+          combinatorType = TokenKind.COMBINATOR_GREATER;
+        }
         break;
       case TokenKind.TILDE:
         _eat(TokenKind.TILDE);
         combinatorType = TokenKind.COMBINATOR_TILDE;
         break;
+      case TokenKind.SLASH:
+        // Parse /deep/
+        _eat(TokenKind.SLASH);
+        var ate = _maybeEat(TokenKind.IDENTIFIER);
+        var tok = ate ? _previousToken : _peekToken;
+        if (!(ate && tok.text == 'deep')) {
```

Comment:
Do you want to support /deep/ ?  I noticed its no longer mentioned in the spec https://drafts.csswg.org/css-scoping-1/#deep-combinator only >>> is the deep combinator.

If you need to support 'deep' then all keywords/reserved words e.g., 'deep' should be in the TokenKind not hard coded in the parser.  Once CSS exposes a name its usually a good chance they'll add more reserved names.  You might do something like:

static const List<Map<String, dynamic>> COMBINATOR_NAMES = const [
    const {'type': TokenKind.COMBINATOR_DEEP, 'value': 'deep'},
];

  static int matchCombinatorNames(String text, int offset, int length) {
    return matchList(COMBINATOR_NAMES, 'type', text, offset, length);
  }

then the parser would have this code:

case TokenKind.SLASH:
     // Parse /deep/
     _eat(TokenKind.SLASH);
     if (_peekIdentifier()) {
        // Is it a directive?
        var combinatorName = _peekToken.text;
        var combinatorLen = directive.length;
        tokId = TokenKind.matchCombinatorNames(combinatorName, 0, combinatorLen);
...